include/deal.II/matrix_free/matrix_free.h

```diff
@@ -684,6 +684,18 @@ class MatrixFree : public Subscriptor
   copy_from(
     const MatrixFree<dim, Number, VectorizedArrayType> &matrix_free_base);
 
+  /**
+   * Refreshes the geometry data stored in the MappingInfo fields when the
+   * underlying geometry has changed (e.g. by a mapping that can deform
+   * through a change in the spatial configuration like MappingFEField)
+   * whereas the topology of the mesh and unknowns have remained the
+   * same. Compared to reinit(), this operation only has to re-generate the
+   * geometry arrays and can thus be significantly cheaper (depending on the
+   * cost to evaluate the geometry).
+   */
+  void
+  update_mapping(const Mapping<dim> &mapping);
```

Comment:
I am wondering if you could add a version without `mapping` input argument. In this case, the old mapping (which is stored in `MappingInfo`) could be used. 

But maybe, I am just not sure how mapping  is used in this context. Is a new mapping create after each let's say timestep or is an existing mapping updated by some delta xs. In the latter case, it would be odd if the mapping has to handed in again...

Reply:
It is true that one could update the underlying data stored in `mapping` and let `MappingInfo` figure out what should happen, given that it holds a pointer to `mapping`. This would be the typical case when we have a `MappingFEField` or `MappingQCache` where the underlying vector/points change. However, we are not bound to this and we could in principle pass in another `Mapping` object altogether.

My motivation was to explicitly state where the updated information comes from, so by passing it in again from where the object lives I force the user to think about it when calling `MatrixFree::update_mapping()` - after all, there must have been an update to it recently to force the update. Or do you see a case where I would call this update functionality from a different place to where the `Mapping` lives?

Reply:
> Or do you see a case where I would call this update functionality from a different place to where the Mapping lives?

You are probably right. This function is probably called somewhere (same class but different method) where the user also has called `MatrixFree::reinit()`...

> My motivation was to explicitly state where the updated information comes from, so by passing it in again from where the object lives I force the user to think about it

This is a good point! Normally I create a mapping object directly before `MatrixFree::reinit`. If I would call `MatrixFree::update_mapping()` without argument there might be the case that the mapping object is already out of scope...